build/includes/allocation.mk

```diff
@@ -0,0 +1,70 @@
+# Copyright 2019 Google LLC All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+#         ____  ____   ____   _____           _ _	
+#    __ _|  _ \|  _ \ / ___| |_   _|__   ___ | (_)_ __   __ _	
+#   / _` | |_) | |_) | |       | |/ _ \ / _ \| | | '_ \ / _` |	
+#  | (_| |  _ <|  __/| |___    | | (_) | (_) | | | | | | (_| |	
+#   \__, |_| \_\_|    \____|   |_|\___/ \___/|_|_|_| |_|\__, |	
+#   |___/                                               |___/ 
+
+build_allocation_base_version = $(call sha,$(build_path)/build-sdk-images/tool/base/Dockerfile)
+build_allocation_base_tag = agones-build-allocation-base:$(build_allocation_base_version)
+
+# Calculate sha hash of sha hashes of all files in a specified ALLOCATION_FOLDER
+build_allocation_version = $(call sha_dir,$(build_path)/build-allocation-images/$(ALLOCATION_FOLDER)/*)
+build_allocation_base_remote_tag = $(REGISTRY)/$(build_allocation_base_tag)
+build_allocation_prefix = agones-build-allocation-
+grpc_release_tag = v1.16.1
+allocation_build_folder = build-allocation-images/
+ALLOCATION_FOLDER ?= go
+COMMAND ?= gen
+ALLOCATION_IMAGE_TAG=$(build_allocation_prefix)$(ALLOCATION_FOLDER):$(build_allocation_version)
+
+.PHONY: gen-allocation-grpc 
+
+# Generates grpc server and client for a single allocation, use ALLOCATION_FOLDER variable to specify the allocation folder.
+gen-allocation-grpc: COMMAND := gen
+gen-allocation-grpc: run-allocation-command
```

Comment:
Should this run the container with gen.sh? Do we need all the searching functionality in `run-allocation-command`?

In this particular instance, it feels like it might be overkill? WDYT? Simpler may be better?

Reply:
I simplified it.